src/core/ext/filters/http/server/http_server_filter.cc

```diff
@@ -131,22 +131,23 @@ static grpc_error* hs_filter_incoming_metadata(grpc_call_element* elem,
   static const char* error_name = "Failed processing incoming headers";
 
   if (b->idx.named.method != nullptr) {
-    if (grpc_mdelem_eq(b->idx.named.method->md, GRPC_MDELEM_METHOD_POST)) {
+    if (grpc_mdelem_eq_static(b->idx.named.method->md,
+                              GRPC_MDELEM_METHOD_GET)) {
+      *calld->recv_initial_metadata_flags |=
+          GRPC_INITIAL_METADATA_CACHEABLE_REQUEST;
+      *calld->recv_initial_metadata_flags &=
+          ~GRPC_INITIAL_METADATA_IDEMPOTENT_REQUEST;
+    } else if (grpc_mdelem_eq_static(b->idx.named.method->md,
+                                     GRPC_MDELEM_METHOD_POST)) {
```

Comment:
POST is the common case here, which is why it was first.  I suspect the re-ordering is going to be sub-optimal.

Reply:
Sure. I do find that surprising (which is why I rearranged it) but I can change it back.

...So most RPC calls (are expected to) make state changes?

Reply:
Changed back.

Reply:
Please push the change before resolving.

Reply:
> ...So most RPC calls (are expected to) make state changes?

It's not actually a question of whether the call makes state changes; it's really a question of how we encode RPCs on the wire.  POST is the expected form for gRPC, as per https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md.  The GET and PUT alternatives here were added later to support cacheable and idempotent requests, but neither of those features are really first-class citizens in our code.